doc/server_side_movement.md

```diff
@@ -0,0 +1,113 @@
+# Server side movement validation
+
+(This document is expected to be removed before the server side movement
```

Comment:
I say it's better to keep this, it seems like useful documentation for anyone working on this.

Reply:
Good point, that line has been removed.